lib/codecs/src/decoding/format/gelf.rs

```diff
@@ -0,0 +1,216 @@
+    pub const LEVEL: &str = "level";
+
+    /// (optional) (deprecated) Send as additional field instead.
+    pub const FACILITY: &str = "facility";
+
+    /// (optional) (deprecated) The line in a file that caused the error (decimal). Send as additional field instead.
+    pub const LINE: &str = "line";
+
+    /// (optional) (deprecated) The file (with path if you want) that caused the error. Send as additional field instead.
+    pub const FILE: &str = "file";
+}
+pub use gelf_fields::*;
+
+/// Config used to build a `GelfDeserializer`.
+#[derive(Debug, Clone, Default, Deserialize, Serialize)]
+pub struct GelfDeserializerConfig;
+
+impl GelfDeserializerConfig {
+    /// Build the `GelfDeserializer` from this configuration.
+    pub fn build(&self) -> GelfDeserializer {
+        GelfDeserializer::default()
+    }
+
+    /// Return the type of event built by this deserializer.
+    pub fn output_type(&self) -> DataType {
+        DataType::Log
+    }
+
+    /// The schema produced by the deserializer.
+    pub fn schema_definition(&self) -> schema::Definition {
+        schema::Definition::empty()
+            .with_field(VERSION, Kind::bytes(), None)
+            .with_field(HOST, Kind::bytes(), None)
+            .with_field(SHORT_MESSAGE, Kind::bytes(), None)
+            .optional_field(FULL_MESSAGE, Kind::bytes(), None)
+            .optional_field(TIMESTAMP, Kind::timestamp(), None)
+            .optional_field(LEVEL, Kind::integer(), None)
+            .optional_field(FACILITY, Kind::bytes(), None)
+            .optional_field(LINE, Kind::integer(), None)
+            .optional_field(FILE, Kind::bytes(), None)
+            // Every field with an underscore (_) prefix will be treated as an additional field.
+            // Allowed characters in field names are any word character (letter, number, underscore), dashes and dots.
+            // Libraries SHOULD not allow to send id as additional field ( _id). Graylog server nodes omit this field automatically.
+            .unknown_fields(Kind::bytes())
+        //
+    }
+}
+
+/// Deserializer that builds an `Event` from a byte frame containing a GELF log
+/// message.
+#[derive(Debug, Clone)]
+pub struct GelfDeserializer {
+    regex: Regex,
+}
+
+impl Default for GelfDeserializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl GelfDeserializer {
+    /// Create a new GelfDeserializer
+    pub fn new() -> GelfDeserializer {
+        GelfDeserializer {
+            regex: Regex::new(r"^[\w\.\-]*$").unwrap(),
+        }
+    }
+
+    // Builds a LogEvent from the parsed GelfMessage, adhering to GELF spec
+    fn insert_fields_from_gelf(&self, parsed: &GelfMessage) -> vector_core::Result<Event> {
+        let mut event = Event::from(parsed.short_message.to_string());
+        let log = event.as_mut_log();
+
+        // GELF spec defines the version as 1.1 which has not changed since 2013
```

Comment:
I think this is right, but I can't also wait to see what/how this breaks for users/tools doing weird things 😆